cmd/query-tee/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+type Config struct {
+	ServerServicePort  int
+	ServerMetricsPort  int
+	BackendEndpoints   string
+	PreferredBackend   string
+	BackendReadTimeout time.Duration
+	LogLevel           logging.Level
+}
+
+func main() {
+	// Parse CLI flags.
+	cfg := Config{}
+	flag.IntVar(&cfg.ServerServicePort, "server.service-port", 80, "The port where the query-tee service listens to.")
+	flag.IntVar(&cfg.ServerMetricsPort, "server.metrics-port", 9900, "The port where metrics are exposed.")
+	flag.StringVar(&cfg.BackendEndpoints, "backend.endpoints", "", "Comma separated list of backend endpoints to query.")
```

Comment:
Nit: flag package supports multiple values for single parameter (`-backend.endpoint <url1> -backend.endpoint <url2>`), I find that nicer than extra splitting on commas.

Reply:
Yes, I did it. Then Grafana jsonnet doesn't support it (args are a map), so I rolled back to comma separated to be able to easily deploy it in our infra.

Reply:
Right. We could also combine the two approaches :-) But let's not bother with that.